src/mutex.cr

```diff
@@ -2,35 +2,71 @@ require "crystal/spin_lock"
 
 # A fiber-safe mutex.
 class Mutex
+  @m = Atomic(Int32).new(0)
```

Comment:
For posterity sake, could `@m` be renamed to something more descriptive? Thank you.

Reply:
Just replaced with `@state`